_studio/mfx_lib/shared/src/mfx_mpeg2_enc_common_hw.cpp

```diff
@@ -62,7 +62,14 @@ mfxStatus MfxHwMpeg2Encode::QueryHwCaps(VideoCORE* pCore, ENCODE_CAPS & hwCaps,
     ddi.reset( CreatePlatformMpeg2Encoder(pCore) );
     if(ddi.get() == NULL)
         return MFX_ERR_NULL_PTR;
-    mfxStatus sts = ddi->QueryEncodeCaps(hwCaps, codecProfileType);
+
+    mfxU16 codecProfile = par->mfx.CodecProfile;
+    if (codecProfile == MFX_PROFILE_UNKNOWN)
```

Comment:
I would properly define profile at higher levels of the stack (e.g. where QueryHWCaps is called).

Reply:
Fixed